src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const likeState = { ...state };
+  let copyObj = {};
+  const result = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'addProperties':
+        for (const key in actions[i].extraData) {
+          likeState[key] = actions[i].extraData[key];
+        }
+        copyObj = { ...likeState };
+        result.push(copyObj);
```

Comment:
Don't repeat yourself: you use the same code 3 times, move it to the end of the cycle